### PR TITLE
Add support for a Spring Boot custom key alias

### LINKED ISSUE #1843: Spring Ssl.getKeyAlias not used
While digging through spring TLS stuff, I noticed that [Ssl.getKeyAlias](https://github.com/line/armeria/blob/master/spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/Ssl.java#L163) is not referenced anywhere. Not sure what the impact is, but I guess a keystore with multiple aliaseses might not work.

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

The `ssl.keyAlias` configuration in Spring Boot is currently
unsupported.

Modifications:

- Add `CustomAliasKeyManagerFactory` and `CustomAliasX509ExtendedKeyManager`
- Wrap the `KeyManagerFactory` with `CustomAliasKeyManagerFactory` to
  support custom key alias
- Miscellaneous:
  - Add `@Nullable` annotations where necessary

Result:

- Fixes #1843

### CODE REVIEW

**Comment by trustin** [comment]
Will add a test case early next week.

File: spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaSettings.java [hunk 1]
```diff
@@ -66,11 +66,13 @@
         /**
          * IP address to bind to. If not set, will bind to all addresses, e.g. {@code 0.0.0.0}.
          */
+        @Nullable
         private String ip;
 
         /**
          * Network interface to bind to. If not set, will bind to the first detected network interface.
          */
+        @Nullable
```
Comment (ikhoon) [inline spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaSettings.java]: 👍

File: spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/Ssl.java [hunk 2]
```diff
@@ -111,8 +128,9 @@ public void setProvider(SslProvider provider) {
      * needed ("need"). Requires a trust store.
      * @return the {@link ClientAuth} to use
      */
+    @Nullable
     public ClientAuth getClientAuth() {
-        return this.clientAuth;
```
Comment (ikhoon) [inline spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/Ssl.java]: 👍

**Comment by trustin** [comment]
@anuraaga You might want to check this out as well.

File: spring/boot-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/AbstractReactiveWebServerCustomKeyAliasTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                    @Override
+                    protected TrustManager[] engineGetTrustManagers() {
+                        return new TrustManager[] {
+                                new X509TrustManager() {
+                                    @Override
+                                    public void checkClientTrusted(X509Certificate[] chain, String authType) {}
+
+                                    @Override
+                                    public void checkServerTrusted(X509Certificate[] chain, String authType) {
+                                        chosenKeyAlias.set(chain[0].getSubjectX500Principal().getName());
```
Comment (ikhoon) [inline spring/boot-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/AbstractReactiveWebServerCustomKeyAliasTest.java]: 👍

**Pull request merged**